homeassistant/components/qbittorrent/config_flow.py

```diff
@@ -22,6 +40,42 @@ class QBittorrentConfigFlow(config_entries.ConfigFlow, domain=DOMAIN):
 
     def __init__(self):
         """Initialize qbittorrent config flow."""
+        self._data = None
+        self._title = None
+
+    async def async_step_import(self, device_config):
+        """Import a configuration.yaml config."""
+        data = {}
+        errors = {}
+
+        data[CONF_URL] = device_config.get(CONF_URL)
+        data[CONF_USERNAME] = device_config.get(CONF_USERNAME)
+        data[CONF_PASSWORD] = device_config.get(CONF_PASSWORD)
+
+        errors = await validate_input(self.hass, data)
+        if not errors:
+            self._data = data
+            self._title = data[CONF_URL]
+            return await self.async_step_import_confirm()
```

Comment:
In reviewing config flow PRs I haven't seen one that requires a user confirmation step after the import. If you do it without a confirmation then the import can be done silently and without disrupting the user with missing entities when they upgrade HA.

Reply:
hmmm I would leave it like this. This way it is also a heads-up to the user, in case the old yaml config isn't removed.
Each reboot they will get this discovery notification and an error message in the HA logging.
![image](https://user-images.githubusercontent.com/51802836/106519033-fc17d300-64da-11eb-9827-fcda62587a3c.png)